internal/policy/targets.go

```diff
@@ -96,6 +103,58 @@ func UpdateDelegation(targetsMetadata *tuf.TargetsMetadata, ruleName string, aut
 	return targetsMetadata, nil
 }
 
+// ReorderDelegations changes the order of delegations, and the new order is
+// specified in `ruleNames []string`.
+func ReorderDelegations(targetsMetadata *tuf.TargetsMetadata, ruleNames []string) (*tuf.TargetsMetadata, error) {
+	// Create a map of all existing delegations for quick look up
+	rolesMap := make(map[string]tuf.Delegation)
+	// Create a set of current rules in metadata, skipping the allow rule
+	currentRules := set.NewSet[string]()
+	for _, delegation := range targetsMetadata.Delegations.Roles {
+		if delegation.Name == AllowRuleName {
+			continue
+		}
+		rolesMap[delegation.Name] = delegation
+		currentRules.Add(delegation.Name)
+	}
+
+	specifiedRules := set.NewSet[string]()
+	for _, name := range ruleNames {
+		if specifiedRules.Has(name) {
+			return nil, fmt.Errorf("%w: '%s'", ErrDuplicatedRuleName, name)
+		}
+		specifiedRules.Add(name)
+	}
+
+	if currentRules.Len() != specifiedRules.Len() {
```

Comment:
This check is insufficient. If currentRules is `rule-1, rule2` and specifiedRules is `rule-1, rule-3`, we should error. We explicitly need a `set.Equal()` check (which in turn _can_ use .Len() as a first heuristic to return false as two sets cannot be equal if they're of different lengths).

Reply:
Will do!